pkg/utils/kubernetes/health/deployment_test.go

```diff
@@ -313,5 +313,17 @@ var _ = Describe("Deployment", func() {
 			Expect(err).NotTo(HaveOccurred())
 			Expect(ok).To(BeFalse())
 		})
+
+		It("should consider the deployment as updated even though there are still stale pods", func() {
+			for i := 0; i < 2; i++ {
+				p := pod.DeepCopy()
+				p.Status.Reason = "Evicted"
+				Expect(fakeClient.Create(ctx, p)).To(Succeed())
+			}
+
+			ok, err := health.DeploymentHasExactNumberOfPods(ctx, fakeClient, deployment)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(ok).To(BeFalse())
+		})
```

Comment:
Would it not be better to have only one evicted pod being created or have an additional "normal" one so that this succeeds? This test would have yielded the exact same result before the change, i.e. there are two pods, which does not match the exact number of expected pods.